### DISCUSSION TITLE
Which Micro-Python PCF8574 Module Recommend?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16096] [General]
Which Micro-Python PCF8574 Module Recommend?  I have a project where I need to sink a GPIO pin on an external device GND.  I was thinking I could use a PCF8574 breakout board.  As long as I connect the external device GPIO pins to the PCF8574 pins, establish a common GND between the external device and the PCF8574, as I set the PCF8574 pin to LOW, should shunt the remote device GPIO pin to GND.  A simple current sink scenario.  The external device GPIO pins are 3.3v, if they were 5v I think I would be using individual transistors not the PCF8574 board.  So the question is, what would be the recommended MP module to control the PCF8574?  There are several and I cannot seem to tell which is better, more popular, etc.

**Comment by rkompass** [comment]
Take the smallest one (in terms of code size) where the code still seems understandable and test if it works.
The task you want to perform seems rather basic/elementary. Also the I2C interface is established for quite some years by now. So it won't make a difference whether you use older or newer modules.

**Comment by scruss** [comment]
I've been very happy with [mcauser/micropython-pcf8574: MicroPython driver for PCF8574 8-Bit I2C I/O Expander with Interrupt](https://github.com/mcauser/micropython-pcf8574/tree/master), but I've mostly been using it for scanning inputs.

You might not need the expander, though. I've used an output pin toggled from high to low to sink (a very small) current to act as a switch closure. I was triggering [Adafruit Audio FX Sound Board](https://www.adafruit.com/product/2133)s this way. Ground has to be common on all devices

**Comment by jonnor** [comment]
https://github.com/mcauser/micropython-pcf8574 is mip installable and looks all in all quite well maintained.

**Comment by bidrohini** [comment]
I think you need the adafruit-circuitpython-pcf8574 library: https://docs.circuitpython.org/projects/pcf8574/en/stable/ or micropython-pcf8574 by Peter Hinch https://github.com/dhylands/python_lcd/blob/master/README.md

**Reply by scruss** [reply]
Please don't suggest:

1. a CircuitPython module that would not work with MicroPython;
2. a CPython library for operating LCDs (written by Dave Hylands, not Peter Hinch) which, again, won't work with MicroPython.

**Comment by bidrohini** [comment]
You can see the following links.'
https://micropython-pcf8574.readthedocs.io/en/latest/pcf8574.html
https://github.com/mcauser/micropython-pcf8574
https://github.com/xreef/PCF8574_micropython_library
If you decide to make a PCB, you can see here: https://www.pcbway.com/project/shareproject/PCF8574_GPIO_I_O_extender_para_Arduino_ESP32_I2C_Modulo_8_entradas_cf881077.html

**Comment by Jibun-no-Kage** [comment]
My project I am also sinking low current/3.3v pin, to turn on LED in a camera tally status board, where the pin is shorted to ground to turn on the given LED on the status board, stop the sink, to turn off the LED on the status board.
